lua/lualine/components/branch.lua

```diff
@@ -1,34 +1,60 @@
-local async = require('lualine.async')
-
 local git_branch
 
-local get_git_branch = async:new({
-  cmd = 'git branch --show-current',
-  on_stdout = function(_, data)
-    if data then
-      git_branch = data:gsub('\n', '')
-    end
-  end,
-  on_stderr = function (_, data)
-    if data then
-      if data:find("fatal: not a git repository") then
-        git_branch = ''
-      end
+-- os specific path separator
+local sep = package.config:sub(1,1)
+
+-- returns full path to git directory for current directory
+local function find_git_dir()
+  local file_dir = vim.fn.expand('%:p:h') .. ';'
+  local git_dir = vim.fn.finddir('.git', file_dir)
+  local git_file = vim.fn.findfile('.git', file_dir)
+  if #git_file > 0 then git_file = vim.fn.fnamemodify(git_file, ':p') end
```

Comment:
care to explain what this does?

Reply:
I would also rename `file_dir` to `current_working_dir`

Reply:
See the coments. Better to leave explanation for anyone who reads the code.